### PR TITLE
Refactor link steps

### PR DESCRIPTION
**Pull request by EDbarvinsky**

### CODE REVIEW

**Comment by EDbarvinsky** [comment]
WIP. One more step will be replaced with composite after merging of element steps

File: vividus-plugin-web-app/src/main/resources/steps/defaults/link.steps [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+When I click on element located `By.linkUrlPart(<URLpart>)`
+
+Composite: When I click on an image with the tooltip '$tooltipImage'
+When I click on element located `By.xpath(.//a[./img[@alt='<tooltipImage>' or @title='<tooltipImage>']])`
+
+Composite: Then a link with the URL '$URL' and tooltip '$tooltip' does not exist
+Then number of elements found by `By.linkUrl(<URL>)->filter.tooltip(<tooltip>)` is equal to `0`
+
+Composite: Then a link tag with href '$href' exists
+Then number of invisible elements `By.xpath(.//link[@href='<href>'])` is equal to `1`
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/resources/steps/defaults/link.steps]: java implementation searches for all elements (both visible and invisible)

File: vividus-plugin-web-app/src/main/resources/steps/defaults/link.steps [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+Priority: 1
+Then number of elements found by `By.linkText(<text>)->filter.linkUrl(<URL>)` is equal to `1`
+
+Composite: Then a [$state] link with the text '$text' and URL '$URL' exists
+Priority: 1
+Then number of elements found by `By.linkText(<text>)->filter.state(<state>).linkUrl(<URL>)` is equal to `1`
+
+Composite: Then a link with the tooltip '$tooltip' exists
+Priority: 1
+Then number of elements found by `By.xpath(.//a)->filter.tooltip(<tooltip>)` is equal to `1`
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/resources/steps/defaults/link.steps]: `By.xpath(.//a[@title='<tooltip>'])` ?

File: vividus-plugin-web-app/src/main/resources/steps/defaults/link.steps [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+Composite: Then a [$state] link with the tooltip '$tooltip' exists
+Priority: 1
+Then number of elements found by `By.xpath(.//a)->filter.state(<state>).tooltip(<tooltip>)` is equal to `1`
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/resources/steps/defaults/link.steps]: `By.xpath(.//a[@title='<tooltip>'])->filter.state(<state>)` ?

**Pull request merged**